sled-agent/src/bin/sled-agent-sim.rs

```diff
@@ -78,6 +78,7 @@ async fn do_run() -> Result<(), CmdError> {
         storage: ConfigStorage {
             // Create 10 "virtual" U.2s, with 1 TB of storage.
             zpools: vec![ConfigZpool { size: 1 << 40 }; 10],
+            ip: args.sled_agent_addr.ip(),
```

Comment:
At @david-crespo 's suggestion, I'm just re-using whatever IP we supplied for the sled agent here when allocating ports for Crucible too.